cmd/cronosd/cmd/fix-unlucky-tx.go

```diff
@@ -0,0 +1,209 @@
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/spf13/cobra"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmcfg "github.com/tendermint/tendermint/config"
+	tmnode "github.com/tendermint/tendermint/node"
+	sm "github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/state/indexer/sink/psql"
+	"github.com/tendermint/tendermint/state/txindex"
+	"github.com/tendermint/tendermint/state/txindex/kv"
+	tmstore "github.com/tendermint/tendermint/store"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+const ExceedBlockGasLimitError = "out of gas in location: block gas meter; gasWanted:"
+
+// FixUnluckyTxCmd update the tx execution result of false-failed tx in tendermint db
+func FixUnluckyTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "fix-unlucky-tx [blocks-file]",
+		Short: "Fix tx execution result of false-failed tx after v0.7.0 upgrade, \"-\" means stdin.",
+		Long:  "Fix tx execution result of false-failed tx after v0.7.0 upgrade.\nWARNING: don't use this command to patch blocks generated before v0.7.0 upgrade",
```

Comment:
as a safe guard, would it be better to hardcode the upgrade height number and skip the command if the height is below this number?

Reply:
that would complicate the code a little bit though.
To do conditional compile with network build tag, we must create separate modules right?

Reply:
maybe just one for mainnet + have a `--force` flag to skip the safe guard?

Reply:
hmm since we specify the chainId, maybe a simple map chainId -> number would be enough?

Reply:
what would happen if someone try to patch a tx before v0.7.0 upgrade? can it leads to possible state corruption?

Reply:
added a `--min-block-height` as safty guard, with default value of `2693800`.

Reply:
from your comment below, I understand that doing it on block before ` v0.7.0` could lead to state corruption because of the event format?

in that case, having the ability to override `--min-block-height` could be dangerous if bad interpreted, unless there is a reason for node operator to use it?

Reply:
maybe at least, strongly enforce the min-block-height when chain-id `cronosmainnet_25-1` is specified

Reply:
we'll provide an external doc to user: https://github.com/crypto-org-chain/cronos/wiki/Patch-Unlucky-Tx#post-v070-upgrade, the users should follow the procedures in the doc.